bentoml/tracing/jaeger.py

```diff
@@ -101,7 +101,7 @@ def span(
                 token = span_context_var.set(scope.span.context)
                 if request_headers and TRACE_ID_HEADER not in request_headers:
                     tracer.inject(
-                        scope.span.context, Format.HTTP_HEADERS, request_headers,
+                        scope.span.context, Format.HTTP_HEADERS, dict(request_headers),
```

Comment:
I think this will break how the headers are injected between marshal and API server here - since the injected headers dict here will not being used by the aiohttp client that sends the request to downstream service: https://github.com/bentoml/BentoML/blob/master/bentoml/marshal/marshal.py#L358

Looks like this will need a bit of refactoring to actually fix the problem, @bojiang any suggestions?

Reply:
I don't like the `inject` approach in fact. Is this necessary?

Reply:
as https://github.com/opentracing/opentracing-python#outbound-request saiad.
if no any outbound http request. just put a empty dict() also acceptable.
or even remove inject function call.

Reply:
which one is preferd?

Reply:
```suggestion
                        scope.span.context, Format.HTTP_HEADERS, headers_dict,
```

Reply:
done